mathesar/api/serializers/records.py

```diff
@@ -10,8 +14,36 @@ class RecordListParameterSerializer(MathesarErrorMessageMixin, serializers.Seria
 
 
 class RecordSerializer(MathesarErrorMessageMixin, serializers.BaseSerializer):
+    def update(self, instance, validated_data):
+        table = self.context['table']
+        record = table.update_record(instance['id'], validated_data)
+        return record
+
+    def create(self, validated_data):
+        table = self.context['table']
+        try:
+            record = table.create_record_or_records(validated_data)
+        except IntegrityError as e:
+            if type(e.orig) == NotNullViolation:
+                raise database_api_exceptions.NotNullViolationAPIException(
+                    e,
+                    status_code=status.HTTP_400_BAD_REQUEST,
+                    table=table
+                )
+            else:
+                raise database_api_exceptions.MathesarAPIException(e, status_code=status.HTTP_400_BAD_REQUEST)
+        return record
+
     def to_representation(self, instance):
         records = instance._asdict() if not isinstance(instance, dict) else instance
         columns_map = self.context['columns_map']
         records = {columns_map[column_name]: column_value for column_name, column_value in records.items()}
         return records
+
+    def to_internal_value(self, data):
+        columns_map = self.context['columns_map'].inverse
+        data = {columns_map[int(column_id)]: value for column_id, value in data.items()}
+        return data
+
+    def is_valid(self, raise_exception=False):
```

Comment:
Why override this if you're only calling `super`? Please add a comment explaining it.